frontends/web/src/components/backbutton/backbutton.tsx

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2024 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ComponentPropsWithoutRef, useCallback } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useEsc } from '@/hooks/keyboard';
+import { Button } from '@/components/forms/button';
+
+type TBackButton = Omit<ComponentPropsWithoutRef<'button'>, 'onClick'> & {
```

Comment:
shouldn't this be using button.tsx:TProps as you are passing props to the Button component?

Reply:
Yes it probably should, but I tried and could not get it to work, as TButton has a bit a special type for `primary` OR `secondary` OR `transparent`.

And BackButton should always be `secondary` anyway, and never have onClick. So I figured it is not worth bothering extending the type of TButton.

Reply:
why even pass on props in the first place? maybe better just make an explicit list of props that are needed for clarity. Or maybe Pick<> instead of Omit<> for the ones that we actually use and want to pass on.

Reply:
Pushed a commit to simplify BackButton, I think disabled, enableEsc, className and children are sufficient.

Children is probably not really necessary, but I like the flexibility so that the parent component can decide to use button.back or "cancel" abort etc. depending on the view/popup